### PR TITLE
Remove a few dependencies to UIKit in SwiftUI's CaseStudies

### PR DESCRIPTION
**Pull request by tgrapperon**
A PR about TCA's the `OpenURL` `Dependency` reminded me that we can use the built-in `@Environment(\.openURL)` in `CaseStudies`.

### CODE REVIEW

File: Examples/CaseStudies/SwiftUICaseStudies/02-Effects-Basics.swift [hunk 1]
```diff
@@ -138,7 +139,7 @@ struct EffectsBasicsView: View {
 
         Section {
           Button("Number facts provided by numbersapi.com") {
-            UIApplication.shared.open(URL(string: "http://numbersapi.com")!)
+            openURL(URL(string: "http://numbersapi.com")!)
```
Comment (mbrandonw) [inline Examples/CaseStudies/SwiftUICaseStudies/02-Effects-Basics.swift:142]:
```suggestion
            self.openURL(URL(string: "http://numbersapi.com")!)
```

File: Examples/CaseStudies/SwiftUICaseStudies/02-Effects-Cancellation.swift [hunk 2]
```diff
@@ -104,7 +105,7 @@ struct EffectsCancellationView: View {
 
         Section {
           Button("Number facts provided by numbersapi.com") {
-            UIApplication.shared.open(URL(string: "http://numbersapi.com")!)
+            openURL(URL(string: "http://numbersapi.com")!)
```
Comment (mbrandonw) [inline Examples/CaseStudies/SwiftUICaseStudies/02-Effects-Cancellation.swift:108]:
```suggestion
            self.openURL(URL(string: "http://numbersapi.com")!)
```

**Review by mbrandonw — APPROVED** [review]
Awesome, thanks!

**Comment by tgrapperon** [comment]
I always forget these explicit `self`! Thanks for fixing them.

**Pull request merged**